graph/pull.go

```diff
@@ -149,7 +149,12 @@ func (s *TagStore) pullRepository(r *registry.Session, out io.Writer, repoInfo *
 	errors := make(chan error)
 
 	layersDownloaded := false
-	for _, image := range repoData.ImgList {
+	imgIDs := []string{}
+	defer func() {
+		s.graph.Release(imgIDs, imgIDs)
```

Comment:
Let's use the same method we did with the builder here. Generate a random ID for this Pull Session and we can just call

```go
defer func() {
    s.graph.Release(imgIDs, pullSessionID)
}()
```

This should make the rest of the retain/release calls less difficult to reason about if every retain is called with `pullSessionID` rather than some other image ID.